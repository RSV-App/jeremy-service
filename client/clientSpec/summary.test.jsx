import React from 'react';
import renderer from 'react-test-renderer';
import Summary from '../src/components/Summary.jsx';

it('renders the summary sub-component', () => {
  const summary = {
    location: 'West Ernieville',
    noise: 'Average',
    recommendPercent: 36,
    valueRating: '4',
    averageOverall: '0.4',
    averageFood: '4.5',
    averageAmbience: '4.5',
    averageService: '5'
  };
  const starPercentages = [10, 30, 20, 35, 5];

  const tree = renderer
    .create(<Summary summary={summary} totalReviews={5} starPercentages={starPercentages} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
