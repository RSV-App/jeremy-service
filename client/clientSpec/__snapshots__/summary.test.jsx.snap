// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the summary sub-component 1`] = `
<div
  className="summary"
>
  <div
    className="header"
  >
    <h4>
      What 
      5
       People Are Saying
    </h4>
  </div>
  <div
    className="sub-header"
  >
    <p>
      Overall ratings and reviews
    </p>
  </div>
  <div
    className="center"
  >
    <div
      className="left"
    >
      <p
        className="disclaimer"
      >
        Reviews can only be made by diners who have eaten at this restaurant
      </p>
      <div
        className="overallStars"
      >
        <div
          className="summary-stars"
        >
          <span
            className="review-star-blank"
          />
          <span
            className="review-star-blank"
          />
          <span
            className="review-star-blank"
          />
          <span
            className="review-star-blank"
          />
          <span
            className="review-star-blank"
          />
        </div>
        <p>
          0.4
           based on recent ratings
        </p>
      </div>
      <div
        className="overallRatings"
      >
        <div
          className="food"
        >
          <p
            className="rating"
          >
            4.5
          </p>
          <p>
            Food
          </p>
        </div>
        <div
          className="service"
        >
          <p
            className="rating"
          >
            5
          </p>
          <p>
            Service
          </p>
        </div>
        <div
          className="ambience"
        >
          <p
            className="rating"
          >
            4.5
          </p>
          <p>
            Ambience
          </p>
        </div>
        <div
          className="value"
        >
          <p
            className="rating"
          >
            4
          </p>
          <p>
            Value
          </p>
        </div>
      </div>
      <div
        className="noise"
      >
        <div
          className="noise-icon"
        />
        <span>
          Noise · 
          Average
        </span>
      </div>
      <div
        className="recommend"
      >
        <div
          className="recommend-icon"
        />
        <span>
          <strong>
            36
            % of people
          </strong>
           would recommend it to a friend
        </span>
      </div>
    </div>
    <div
      className="right"
    >
      <div
        className="bars"
      >
        <div
          className="bar"
        >
          <span>
            NaN
          </span>
          <div
            className="bar-container"
            id="barNaN"
          >
            <div
              className="progress"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </div>
        </div>
        <div
          className="bar"
        >
          <span>
            NaN
          </span>
          <div
            className="bar-container"
            id="barNaN"
          >
            <div
              className="progress"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </div>
        </div>
        <div
          className="bar"
        >
          <span>
            NaN
          </span>
          <div
            className="bar-container"
            id="barNaN"
          >
            <div
              className="progress"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </div>
        </div>
        <div
          className="bar"
        >
          <span>
            NaN
          </span>
          <div
            className="bar-container"
            id="barNaN"
          >
            <div
              className="progress"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </div>
        </div>
        <div
          className="bar"
        >
          <span>
            NaN
          </span>
          <div
            className="bar-container"
            id="barNaN"
          >
            <div
              className="progress"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="summary-bottom"
  >
    <div
      className="link"
    >
      <a>
        Best Restaurants in 
        West Ernieville
         ›
      </a>
    </div>
  </div>
</div>
`;
